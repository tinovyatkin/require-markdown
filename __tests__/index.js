'use strict';

const requireMarkdown = require('../');

describe('require-markdown', () => {
  test('load README', () => {
    const rm = requireMarkdown();
    const res = rm('../README.md');
    expect(res).toMatchSnapshot();
  });
});
