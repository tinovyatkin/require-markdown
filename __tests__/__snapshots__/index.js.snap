// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`require-markdown load README 1`] = `
"<h1>require-markdown</h1>
<p>Loads Markdown files as compiled HTML string synchronously (as Node <code>require</code>)</p>
<h2>Motivation</h2>
<p>This module suppose to be used in build systems, etc, where you prefer to store texts as Markdown files. Uses <a href=\\"https://github.com/markdown-it/markdown-it\\">markdown-it</a> for rendering with emoji, subscript, superscript, comments stripping and abbr plugins enabled.</p>
<h2>Usage</h2>
<pre><code class=\\"language-js\\">const requireMarkdown = require(&quot;require-markdown&quot;)(
  (markdownItOptions = { typographer: true })
);

const htmlString = requireMarkdown(&quot;../README.md&quot;); // =&gt; '&lt;h1&gt;require-markdown&lt;/h1&gt;' ....
</code></pre>
"
`;
